src/ansys/fluent/core/session.py

```diff
@@ -348,3 +376,61 @@ def root(self):
             if self._settings_root is None:
                 self._settings_root = settings_get_root(flproxy=self._settings_service)
             return self._settings_root
+
+
+class _Uploader:
+    """Instantiates a file uploader and downloader to have a seamless file
+    reading / writing in the cloud particularly in Ansys lab . Here we are
+    exposing upload and download methods on session objects. These would be no-
+    ops if PyPIM is not configured or not authorized with the appropriate
+    service. This will be used for internal purpose only.
+
+    Attributes
+    ----------
+    pim_instance: PIM instance
+        Instance of PIM which supports upload server services.
+
+    file_service: Client instance
+        Instance of Client which supports upload and download methods.
+
+    Methods
+    -------
+    upload(
+        file_path, remote_file_name
+        )
+        Upload a file to the server.
+
+    download(
+        file_name, local_file_path
+        )
+        Download a file from the server.
+    """
+
+    def __init__(self, pim_instance):
+        self.pim_instance = pim_instance
+
+        try:
+            upload_server = self.pim_instance.services["http-simple-upload-server"]
+        except AttributeError:
+            LOG.error("PIM is not installed or not authorized.")
+        except KeyError:
+            self.file_service = None
+        else:
+            from simple_upload_server.client import Client
+
+            self.file_service = Client(
+                token="token", url=upload_server.uri, headers=upload_server.headers
+            )
+
+    def upload(self, file_path: str, remote_file_name: str = None):
+        """Uploads a file on the server."""
+        expanded_file_path = os.path.expandvars(file_path)
+        upload_file_name = remote_file_name or os.path.basename(expanded_file_path)
+        self.file_service.upload_file(expanded_file_path, upload_file_name)
+
+    def download(self, file_name: str, local_file_path: str = None):
+        """Downloads a file from the server."""
+        if self.file_service.file_exist(file_name):
+            self.file_service.download_file(file_name, local_file_path)
+        else:
+            raise FileNotFoundError("Uploaded remote file name is different.")
```

Comment:
Can we just say that the remote file does not exist?

Reply:
Yes, we can. Done.